### ISSUE TITLE
Section folding not working under TaskWiki

### ISSUE THREAD

**Issue by paralin** [tbabej/taskwiki#147]
In VimWiki regularly, I can fold section headers with `zc`. But in a TaskWiki buffer, I cannot fold section headers - I just get the error "No fold found."

**Comment by tbabej** [comment]
Taskwiki overrides Vimwiki folding function with its own implementation, which only works for Mediawiki styled headers. You might want to comment out the line 72 of `taskwiki/after/syntax/vimwiki.vim`

**Comment by paralin** [comment]
Interesting. Would be good to have a toggle for that.

**Comment by tbabej** [comment]
Indeed. Or a full blown markdown syntax support.

**Comment by tbabej** [comment]
Added a note about this to parent issue, closing for now.

**Comment by ludenticus** [comment]
> Taskwiki overrides Vimwiki folding function with its own implementation, which only works for Mediawiki styled headers. You might want to comment out the line 72 of `taskwiki/after/syntax/vimwiki.vim`

vimwiki.vim has now (2021-04-08) only 54 lines... What's the line you have to comment out?

**Comment by tbabej** [comment]
Hi @ludenticus, I'm wondering - are you on latest version? The section folding behaviour should no longer change since 7b72ee47b993e4389fe4160091314fccf4845d6b

**Comment by ludenticus** [comment]
Hi, @tbabej. Thanks for your answer. I have latest version *via* Vundle > PluginUpdate. If I don't load taskwiki, vimwiki folds as it is supposed to do. I use `vimwiki` + `vim-taskwarrior` + `vim-markdown-folding`. Viewports and automatic task detection within the wiki work as intended. I've tried `vimrc` with and without `let g:taskwiki_markup_syntax = 'markdown'`, but found no difference.

**Comment by ludenticus** [comment]
Using `taskwiki`, `set foldmethod?` reports `syntax`. Without `taskwiki`, `set foldmethod?` reports `expr`. It is clearly overriding this option.

**Comment by tbabej** [comment]
What does `set foldtext?` say for you?

**Comment by ludenticus** [comment]
`set foldtext?` reports `foldtext=taskwiki#FoldText()`. Otherwise (`taskwiki` disabled), `foldtext=foldtext()`

**Comment by maneeshsethi** [comment]
Hi there, I get a 'foldtext=syntax', however, I still get a 'no fold found' error. Any clue why?

**Comment by SemyonDunaev** [comment]
Looks like let g:taskwiki_dont_fold = 1 not working.
I tried everything that was advised here, but still did not solve the folding with markdown.

**Comment by folofjc** [comment]
I agree, `let g:taskwiki_dont_fold = 1` is not working. I am still getting that taskwiki is setting the fold to syntax when vimwiki is set to custom.

The problem is that if you use markdown with vimwiki and then follow the instructions [starting here](https://github.com/vimwiki/vimwiki/blob/63af6e72dd3fa840bffb3ebcb8c96970c02e0913/doc/vimwiki.txt#L1548), taskwiki overwrites it.

**Comment by mediapathic** [comment]
Bumping this to say that I'm also still having this problem. Setting `g:vimwiki_folding=custom` and `g:taskwiki_dont_fold=1` does nothing, taskwarrior still takes over folding (incorrectly). Uninstalling taskwiki is currently the only way I have to get folding to work correctly.

**Comment by mauromotion** [comment]
Same here. I'm using it with Neovim v0.9.2. with markdown syntax.

My VimWiki index page is broken when taskwiki is active: I have four links to four categories of my notes and if I press enter on any of them a new split opens in the bottom which says: 

```
Filter used: 
Defaults used:

taskwiki.98092 [-]
```
I need to quit the split and I can't enter my links anymore.

Also each and every haeding in any page is folded by default when taskwiki is installed. And I haven't used any foldings before in my VimWIki notes. 

Adding `g.vim.taskwiki_dont_fold=1` has no effect. 

As of now I have to uninstall taskwiki to be able to use VimWiki at all.

**Comment by Jimmy2027** [comment]
Same issue for me. Would be amazing to have taskwiki work with markdown files!

**Comment by mauromotion** [comment]
> Same issue for me. Would be amazing to have taskwiki work with markdown files!

I don't know if it's the same issue but I managed to find a "solution". My issue was that the links in my index.md had headings (##). Once I removed those headings taskwiki didn't interfere with the page and it just worked. Also, I managed to disable the folding, the command I was using was wrong. 

Anyway here's my vimwiki.lua including taskwiki as a dependency with all the settings if it helps: https://github.com/mauromotion/dotfiles/blob/main/nvim_mauro/lua/plugins/utilities/vimwiki.lua

**Comment by Jimmy2027** [comment]
Right, this works for me as well, thanks a lot @mauromotion 🙏
```
let g:taskwiki_dont_preserve_folds = "yes"
let g:taskwiki_dont_fold = "yes"
```
It even works with ## headings
